tests/notnull_tests.cpp

```diff
@@ -80,20 +80,20 @@ SUITE(NotNullTests)
 	p = q; // not possible to implicitly convert MyBase* to MyDerived*
 
         not_null<Unrelated*> r = p;
-        not_null<Unrelated*> s = reinterpret_cast<Unrelated*>(p);
+        not_null<Unrelated*> s(reinterpret_cast<Unrelated*>(p));
 #endif
-        not_null<Unrelated*> t = reinterpret_cast<Unrelated*>(p.get());
+        not_null<Unrelated*> t(reinterpret_cast<Unrelated*>(p.get()));
         CHECK((void*)p.get() == (void*)t.get());
     }
 
     TEST(TestNotNullAssignment)
     {
         int i = 12;
-        not_null<int*> p = &i; 
+        not_null<int*> p(&i); 
```

Comment:
Do we need a make_not_null factory ?
```
void f(not_null<int*>);
f(make_not_null(&i))
```

Wondering if in the particular case of `not_null<T*>` it couldn't be copy-constructed and assigned from a `T&`.

```
f(i); // implicit conversion from T& to `not_null<T*>`
```